cmp/grid/Grid.ts

```diff
@@ -867,3 +867,19 @@ export class GridLocalModel extends HoistModel {
         }
     };
 }
+
+/**
+ * When a `Grid` context menu is open at the same time as a BP `Overlay2` with `enforceFocus`, the
+ * context menu will lose focus, causing menu items not to highlight on hover. By conditionally
+ * stopping the focus event from propagating, we can prevent this behavior.
+ */
```

Comment:
Nice working finding that this fixes the issue.

To make it super safe can we:
only add the listener on grid render on model instantiation, maybe with a hook, and then remove the listener on gridmodel destroy?

I think that if the handler is a named function rather than a fat arrow anonymous function, it is easier to remove with removeEventListener.

Reply:
Good point, and yes - that's exactly why I added it the way that I did, but I think I've got something more elegant now (with a little more code complexity, but encapsulated in `GridLocalModel` which I think is better)